testing/core/variable_test.py

```diff
@@ -40,3 +41,12 @@ def func(component):
         self.assertTrue(x2.uuid == m.x.uuid)
         self.assertTrue(x2.shape == m.x.shape, (x2.shape, m.x.shape))
         self.assertTrue(y in m)
+
+    def test_array_variable_shape(self):
```

Comment:
Can you add a few tests here that test when the value's shape and the passed in shape disagree?

Something like:
```python
mxnet_array_shape = (3, 2)
variable_array_shape = (3,)
try:
    variable = mf.Variable(value=mx.nd.zeros(shape=mxnet_array_shape), shape=variable_array_shape)
catch e:
    return
self.assertFalse(False, "Should have thrown an exception for conflicting variable shapes.")
```

Reply:
Sure, I was planning to add some assertRaises Exception type checks but missed. Will do :)